extensions/rocksdb-repos/FlowFileLoader.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "FlowFileLoader.h"
+
+#include <span>
+#include <memory>
+#include <string>
+#include <vector>
+#include <utility>
+
+#include "logging/LoggerConfiguration.h"
+#include "FlowFileRecord.h"
+
+namespace org::apache::nifi::minifi {
+
+FlowFileLoader::FlowFileLoader()
+  : logger_(core::logging::LoggerFactory<FlowFileLoader>::getLogger()) {}
+
+FlowFileLoader::~FlowFileLoader() {
+  stop();
+}
+
+void FlowFileLoader::initialize(gsl::not_null<minifi::internal::RocksDatabase *> db, std::shared_ptr<core::ContentRepository> content_repo) {
+  db_ = db;
+  content_repo_ = std::move(content_repo);
+}
+
+std::future<FlowFileLoader::FlowFilePtrVec> FlowFileLoader::load(std::vector<SwappedFlowFile> flow_files) {
+  auto promise = std::make_shared<std::promise<FlowFilePtrVec>>();
+  std::future<FlowFilePtrVec> future = promise->get_future();
+  utils::Worker<utils::TaskRescheduleInfo> task{
+    std::bind(&FlowFileLoader::loadImpl, this, std::move(flow_files), std::move(promise)),
+    "",  // doesn't matter that tasks alias by name, as we never actually query their status or stop a single task
+    std::make_unique<utils::ComplexMonitor>()};
+  std::future<utils::TaskRescheduleInfo> dummy_future;
+  thread_pool_.execute(std::move(task), dummy_future);
+  return future;
```

Comment:
Shouldn't this return a future that's resolved when the flow file is loaded? I find the current approach confusing.

Reply:
added comment

so the Worker wants to determine if it should rerun based on the functors return value + the `run_determinant` object, and then it returns that result, so we would need to write a custom `run_determinant` to handle a return value containing an actual value not just the `TaskRescheduleInfo`

Reply:
I feel like this is hacking around the added complexity of using ThreadPool. Why not use utils::FifoExecutor? I may be biased, since I wrote FifoExecutor, but it's meant to solve this exact problem: You give it a function of T, and it gives back a future of T.